pyproject.toml

```diff
@@ -1,10 +1,81 @@
 [build-system]
-requires = ['setuptools >= 40.9.0', 'wheel', 'packutil']
-build-backend = 'setuptools.build_meta'
+requires = ["poetry-core>=1.0.0", "poetry-dynamic-versioning"]
+build-backend = "poetry.core.masonry.api"
+
+[tool.poetry]
+name = "eko"
+version = "0.0.0"
+description = "Evolution Kernel Operator"
+authors = [
+  "A. Candido <alessandro.candido@mi.infn.it>",
+  "F. Hekhorn <felix.hekhorn@mi.infn.it>",
+  "G. Magni <gmagni@nikhef.nl>",
+]
+classifiers = [
+  "Programming Language :: Python",
+  "Programming Language :: Python :: 3",
+  "Topic :: Scientific/Engineering",
+  "Topic :: Scientific/Engineering :: Physics",
+  "License :: OSI Approved :: GNU General Public License v3 (GPLv3)",
+]
+repository = "https://github.com/N3PDF/eko"
+include = ["doc/source/img/Logo.png"]
+
+[tool.poetry.dependencies]
+python = "^3.8,<3.10"
+numpy = "^1.20.0"
+scipy = "^1.7.3"
+PyYAML = "^6.0"
+lz4 = "^3.1.10"
+numba = "^0.54.1"
+# docs dependencies (for readthedocs, https://github.com/readthedocs/readthedocs.org/issues/4912#issuecomment-664002569)
+Sphinx = { version = "^4.3.2", optional = true }
+sphinx-rtd-theme = { version = "^1.0.0", optional = true }
+sphinxcontrib-bibtex = { version = "^2.4.1", optional = true }
+
+[tool.poetry.dev-dependencies]
+black = "^21.12b0"
+isort = "^5.10.1"
+pylint = "^2.12.2"
+pre-commit = "^2.16.0"
+pytest = "^6.2.5"
+pytest-cov = "^3.0.0"
+pytest-env = "^0.6.2"
+# benchmark
+ekomark = { path = "benchmarks/", develop = true }
+Sphinx = "^4.3.2"
+sphinx-rtd-theme = "^1.0.0"
+sphinxcontrib-bibtex = "^2.4.1"
+
+[tool.poetry.extras]
+docs = ["sphinx", "sphinx-rtd-theme", "sphinxcontrib-bibtex"]
+
+[tool.poetry-dynamic-versioning]
+enable = true
+vcs = "git"
+style = "semver"
+dirty = true
+
+[tool.poetry-dynamic-versioning.substitution]
+files = ["src/eko/version.py"]
+
+[tool.poe.tasks]
+# until https://github.com/numba/numba/pull/5660 is confirmed we need to deactivate numba prior running
+test = { shell = "NUMBA_DISABLE_JIT=1 pytest tests" }
+lint = "pylint src/*/*.py -E"
+lint-warnings = "pylint src/*/*.py --exit-zero"
+sandbox = "python benchmarks/runners/sandbox.py"
+# navigator = "navigator"
+black = "black ."
+isort = "isort . --skip env"
+format = ["black", "isort"]
+docs = { "shell" = "cd doc; make html" }
```

Comment:
VS code is comlaining 
```
[{
	"resource": "/home/felix/Physik/N3PDF/EKO/eko/pyproject.toml",
	"owner": "_generated_diagnostic_collection_name_#2",
	"severity": 8,
	"message": "Expected \"}\", [ \\t] or [A-Za-z0-9_\\-] but \"\\\"\" found.",
	"source": "Toml Parser",
	"startLineNumber": 72,
	"startColumn": 11,
	"endLineNumber": 72,
	"endColumn": 12
}]```

are you sure this is working?

Reply:
Have you tried to run `poe docs`, or even simply [`poetry check`](https://python-poetry.org/docs/cli/#check), or a `toml` loader?

I autoformat everything, and that should be valid.
For reference, my current `toml` formatter is: [`taplo`](https://taplo.tamasfe.dev/).

Reply:
well, seems to work ... maybe this is a bug in https://github.com/bungcip/better-toml - I switched to the taplo implementation as well (as the other one doesn't seem to be maintained any longer ...)

Reply:
Yeah, unfortunately `toml` is being adopted, but the adoption is rather recent, and TOML 1.0 is even more recent, thus we're lacking tooling.
(Even though `taplo` is quite good, and rather a complete toolbox, but for some reason it doesn't rank high in searches).

Actually, major uses case of `toml` are `Cargo.toml` and `pyproject.toml` themselves (within a few other majors, but I'd say a bit less relevant). In this [official list](https://github.com/toml-lang/toml/wiki#projects-using-toml) all tools related to python are listed because of `pyproject.toml` (black, flit, poetry, pip, tox, pipenv, isort, ...).

Reply:
All in all, I'd say that TOML is pretty good, but young. It's the younger and improved version of INI (with the clear advantage of having a standard, excluding which INI was already rather good for configs).

Nevertheless, keep in mind that even TOML has pitfalls, and [not everyone likes it](https://hitchdev.com/strictyaml/why-not/toml/).